internal/controller/nonadminbackup_controller_test.go

```diff
@@ -29,63 +29,65 @@ import (
+	result                       reconcile.Result
+	createVeleroBackup           bool
+	uuidCreatedByReconcile       bool
+	uuidFromTestCase             bool
 }
 
 type nonAdminBackupFullReconcileScenario struct {
 	spec   nacv1alpha1.NonAdminBackupSpec
 	status nacv1alpha1.NonAdminBackupStatus
 }
 
-func buildTestNonAdminBackup(namespace string, spec nacv1alpha1.NonAdminBackupSpec) *nacv1alpha1.NonAdminBackup {
+func buildTestNonAdminBackup(nonAdminNamespace string, nonAdminName string, spec nacv1alpha1.NonAdminBackupSpec) *nacv1alpha1.NonAdminBackup {
 	return &nacv1alpha1.NonAdminBackup{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      testNonAdminBackupName,
-			Namespace: namespace,
+			Name:      nonAdminName,
+			Namespace: nonAdminNamespace,
 		},
 		Spec: spec,
 	}
 }
 
-func checkTestNonAdminBackupStatus(nonAdminBackup *nacv1alpha1.NonAdminBackup, expectedStatus nacv1alpha1.NonAdminBackupStatus, nonAdminNamespaceName string, oadpNamespaceName string) error {
+func checkTestNonAdminBackupStatus(nonAdminBackup *nacv1alpha1.NonAdminBackup, expectedStatus nacv1alpha1.NonAdminBackupStatus, oadpNamespaceName string) error {
 	if nonAdminBackup.Status.Phase != expectedStatus.Phase {
 		return fmt.Errorf("NonAdminBackup Status Phase %v is not equal to expected %v", nonAdminBackup.Status.Phase, expectedStatus.Phase)
 	}
-	if expectedStatus.VeleroBackup != nil {
-		veleroBackupName := expectedStatus.VeleroBackup.Name
-		if expectedStatus.VeleroBackup.Name == placeholder {
-			veleroBackupName = function.GenerateVeleroBackupName(nonAdminNamespaceName, testNonAdminBackupName)
-		}
-		if nonAdminBackup.Status.VeleroBackup.Name != veleroBackupName {
-			return fmt.Errorf("NonAdminBackup Status VeleroBackupName %v is not equal to expected %v", nonAdminBackup.Status.VeleroBackup.Name, veleroBackupName)
-		}
-		veleroBackupNamespace := expectedStatus.VeleroBackup.Namespace
-		if expectedStatus.VeleroBackup.Namespace == placeholder {
-			veleroBackupNamespace = oadpNamespaceName
-		}
-		if nonAdminBackup.Status.VeleroBackup.Namespace != veleroBackupNamespace {
-			return fmt.Errorf("NonAdminBackup Status VeleroBackupNamespace %v is not equal to expected %v", nonAdminBackup.Status.VeleroBackup.Namespace, veleroBackupNamespace)
+
+	if nonAdminBackup.Status.VeleroBackup != nil {
+		if nonAdminBackup.Status.VeleroBackup.NameUUID == "" {
+			return fmt.Errorf("NonAdminBackup Status VeleroBackupName %v is 0 length string", nonAdminBackup.Status.VeleroBackup.NameUUID)
 		}
-		if !reflect.DeepEqual(nonAdminBackup.Status.VeleroBackup.Status, expectedStatus.VeleroBackup.Status) {
-			return fmt.Errorf("NonAdminBackup Status VeleroBackupStatus %v is not equal to expected %v", nonAdminBackup.Status.VeleroBackup.Status, expectedStatus.VeleroBackup.Status)
+
+		if expectedStatus.VeleroBackup != nil {
+			// When there is no VeleroBackup expected Namespace provided, use one that should be result of reconcile loop
```

Comment:
was not easier to understand code with the placeholder approach?

Reply:
I want the input of the test to be in one place and output in another
Placeholder was introducing some external to reconcile logic which wasn't clear to me and also not always real use case scenario, e.g.:

```
			ExpectedStatus: nacv1alpha1.NonAdminBackupStatus{
				Phase: nacv1alpha1.NonAdminBackupPhaseCreated,
				VeleroBackup: &nacv1alpha1.VeleroBackup{
					Name:      placeholder,
					Namespace: placeholder,
					Status:    &velerov1.BackupStatus{},
				},
```

There is not a chance that expected VeleroBackup will contain placeholder as Name or Namespace if the input VeleroBackup object did not have those fields populated, so we introduced some extra unnecessary logic within the test flow.